File: JS/ValidParentheses.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./ValidParentheses.js')

TEST('Starting ValidParentheses test...', (t) => {
    //Test 1
    t.assert(OBJ.isValid("()") === true, "Expect: () = valid")
    //Test 2
    t.assert(OBJ.isValid("()[]{}") === true, "Expect: ()[]{} = valid")
    //Test 3
    t.assert(OBJ.isValid("(]") === false, "Expect: (] = invalid")
    //Test 4
    t.assert(OBJ.isValid("([)]") === false, "Expect: ([)] = invalid")
    //Test 5
    t.assert(OBJ.isValid("{[]}") === true, "Expect: {[]} = valid")
 
    t.end()
})
 

File: GO/ValidParentheses_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestValidParentheses(t *testing.T) {
	tables := []struct {
		ops  string
		want bool
	}{
		{"()", true},
		{"()[]{}", true},
		{"(]", false},
		{"([)]", false},
		{"{[]}", true},
	}

	for _, table := range tables {
		got := isValid(table.ops)

		if got != table.want {
			t.Errorf("Waiting for this %t but the return was this: %t", table.want, got)
		}
	}
}
